supervisor/dbus/network/wireless.py

```diff
@@ -32,14 +35,23 @@ def active(self) -> NetworkWirelessAP | None:
         return self._active
 
     @dbus_connected
-    def request_scan(self) -> Awaitable[None]:
+    async def request_scan(self) -> None:
         """Request a new AP scan."""
-        return self.dbus.Device.Wireless.RequestScan(("a{sv}", {}))
+        await self.dbus.Device.Wireless.RequestScan(("a{sv}", {}))
 
     @dbus_connected
-    def get_all_accesspoints(self) -> Awaitable[Any]:
+    async def get_all_accesspoints(self) -> list[NetworkWirelessAP]:
         """Return a list of all access points path."""
-        return self.dbus.Device.Wireless.GetAllAccessPoints()
+        accesspoints_data = (await self.dbus.Device.Wireless.GetAllAccessPoints())[0]
+        accesspoints = [NetworkWirelessAP(ap_obj) for ap_obj in accesspoints_data]
+
+        for err in await asyncio.gather(
+            *[ap.connect() for ap in accesspoints], return_exceptions=True
+        ):
+            if err:
+                _LOGGER.warning("Can't process an AP: %s", err)
+
+        return accesspoints
```

Comment:
This actually isn't a functional change although it looks like one. The only caller of this method today is here:
https://github.com/home-assistant/supervisor/blob/01e24a3e74c54c4b93e4c87196d1a5ebf8d0fb42/supervisor/host/network.py#L272-L293

It works identically even down to the logged warning. Except now the processing of setting object paths into connected `NetworkWirelessAP` objects occurs here and `scan_wifi` just uses the successfully connected ones. I felt it was better to move knowledge of the dbus objects in here and just return python objects.